ui/src/lib/yjs/useYjsStore.ts

```diff
@@ -28,67 +28,77 @@ export default ({
 }) => {
   const { toast } = useToast();
   const t = useT();
-
   const [currentProject] = useCurrentProject();
   const { createDeployment, useUpdateDeployment } = useDeployment();
-
-  const yWebSocketRef = useRef<WebsocketProvider | null>(null);
-  useEffect(() => () => yWebSocketRef.current?.destroy(), []);
-
+  const managerRef = useRef<SocketYjsManager | null>(null);
   const [undoManager, setUndoManager] = useState<Y.UndoManager | null>(null);
 
-  const [{ yWorkflows, currentUserClientId, undoTrackerActionWrapper }] =
-    useState(() => {
-      const yDoc = new Y.Doc();
-      const { websocket, websocketToken } = config();
-      if (workflowId && websocket && websocketToken) {
-        yWebSocketRef.current = new WebsocketProvider(
-          websocket,
-          workflowId,
-          yDoc,
-          { params: { token: websocketToken } },
-        );
-      }
+  // Initialize store and connect
+  const [{ yWorkflows, currentUserClientId }] = useState(() => {
+    const doc = new Y.Doc();
+    const yWorkflows = doc.getArray<YWorkflow>("workflows");
 
-      const yWorkflows = yDoc.getArray<YWorkflow>("workflows");
+    // Initialize with default workflow if empty
+    if (yWorkflows.length === 0) {
       const yWorkflow = yWorkflowBuilder("main", "Main Workflow");
       yWorkflows.push([yWorkflow]);
+    }
 
-      const currentUserClientId = yDoc.clientID;
-
-      // NOTE: any changes to the yDoc should be wrapped in a transact
-      const undoTrackerActionWrapper = (callback: () => void) =>
-        yDoc.transact(callback, currentUserClientId);
-
-      return { yWorkflows, currentUserClientId, undoTrackerActionWrapper };
-    });
+    return {
+      yWorkflows,
+      currentUserClientId: doc.clientID,
+    };
+  });
 
   useEffect(() => {
-    if (yWorkflows) {
-      const manager = new Y.UndoManager(yWorkflows, {
-        trackedOrigins: new Set([currentUserClientId]), // Only track local changes
-        captureTimeout: 200, // default is 500. 200ms is a good balance between performance and user experience
+    const { websocket: websocketUrl, websocketToken } = config();
+    if (!websocketUrl || !websocketToken || !workflowId || !currentProject?.id)
+      return;
+
+    const doc = managerRef.current?.getDoc() || yWorkflows.doc;
+
+    // Create and setup socket manager
+    if (!doc) return;
+    const manager = new SocketYjsManager(doc);
+    manager
+      .setupSocket({
+        url: websocketUrl,
+        roomId: workflowId,
```

Comment:
The workflowId is probably needing to be changed to projectId. Unless we want a room created and used per workflow/subworkflow. @pyshx What do you think?

Reply:
Ah since we are passing the projectId too, I guess that IS the idea. Right?

Reply:
But each room will have the same Y.array doc with all workflows anyways, so if anything this will just cause some perf issues around changing between canvas... hmmm